### DISCUSSION TITLE
SX1262 AFSK?

### DISCUSSION THREAD

**Discussion by elaidlaw** [jgromes/RadioLib#398] [General]
I recently purchased the SX1262MB2CAS Arduino shield ([link](https://os.mbed.com/components/SX126xMB2xAS/)) and my goal is to use it to transmit AX.25 packets with AFSK modulation. The AX.25 AFSK example doesn't mention that the SX1262 won't work but I don't see anything that indicates it can't. Is there something inherent about the SX1262 that makes AFSK impossible?

If it is possible, it seems like I would need to connect the chip's DIO2 pin to the Arduino, but I don't think the dev board I linked to above has that. Does that check out? If I do need that pin, does anyone know of a SX1262 module that exposes it that I could try using?

Sorry if these are stupid questions, I'm new to the radio world!

**Comment by jgromes** [comment]
SX126x can't do AFSK - the examples explicitly list all modules that can do it, SX126x is not among them.

The reason is that SX126x doesn't support any sort of "direct mode" - FSK Rx/Tx mode in which data is fed to CLK and DATA pins of the module and directly modulated, completely bypassing packet engine. In RadioLib, that's what is used to achieve AFSK.

**Reply by elaidlaw** [reply]
Ah okay that makes sense. Will probably go for the SX1278 then. Is there a module for prototyping that you have used and can recommend as something that will work for AFSK?

**Reply by jgromes** [reply]
Depends on what sort of Arduino you intend to use. If it's something with 5V logic, I would use some module with built-in logic level shifters, I believe Adafruit makes them, but you'll probably find some clones on ebay or aliexpress. If you intend to use 3.3V logic Arduino, you don't need the level shifter and almost any ebay module should work.

I don't really have an exact recommendation, since I usually use a custom shield on Arduino Uno for development purposes.

**Reply by elaidlaw** [reply]
https://www.adafruit.com/product/3073

Should this work? I'm on a pretty tight schedule and all of the SX1278 modules I could find ship from China. Seems like the RFM9x chips are based on SX127x though and therefore have the same functionality? I really appreciate your help

**Reply by jgromes** [reply]
RFM96 is nearly the same chip as SX1278. The only difference is in frequency range, which is 433.0 MHz to 470.0 MHz according to the datasheet.

All RFM9x modules are basically just SX127x in disguise - it's the same silicon, just sold by HopeRF and not Semtech.
